cmd/server-mux.go

```diff
@@ -514,6 +637,11 @@ func (m *ServerMux) Close() error {
 	}
 	m.mu.Unlock()
 
+	if m.tlsLoaded {
+		// Stop tls update routine
+		m.tlsUpdateDoneCh <- struct{}{}
```

Comment:
You would need to close `m.tlsUpdateDoneCh`.  Basically channel closing also triggers done operation ie no need to send `struct{}{}`